File: angular2-tour-of-heroes/app/hero.service.ts
import {Injectable} from 'angular2/core';
import {Hero} from './hero';
import {HEROES} from './mock-heroes';

@Injectable()
export class HeroService {
  getHeroes() {
    return Promise.resolve(HEROES);
  }
  getHero(id) {
    return Promise.resolve(HEROES.filter(i => i.id === id)[0]);
  }
}

File: dtour/lib/hero_service.dart
import 'package:dtour/Hero.dart' show Hero;
import 'package:dtour/mock_heroes.dart' show HEROES;
import 'package:angular2/core.dart';
import 'dart:async';

@Injectable()
class HeroService {
  Future<List<Hero>> getHeroes() {
    return new Future(() => HEROES);
  }
}
